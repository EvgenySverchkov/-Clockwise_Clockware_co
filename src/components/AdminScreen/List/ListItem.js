import React from 'react';
import PropTypes from 'prop-types';

function ListItem({infoObj}){
  let keysArr = Object.keys(infoObj);
  return (
    <>
      {
        keysArr.map((item)=><td className="text-center"
                                style={{wordWrap: "break-word"}}
                                key={item}>{infoObj[item]}</td>)
      }
    </>
  );
}

ListItem.propTypes = {
  infoObj: PropTypes.object
}

export default ListItem;
