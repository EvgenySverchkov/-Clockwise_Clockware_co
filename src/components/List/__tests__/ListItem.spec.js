import React from "react";
import {create} from 'react-test-renderer';

import ListItem from "../ListItem";

it("Test of <ListHeader />", ()=>{
    const mockProps = {
        infoObj: {
            id: 1,
            rating: 5,
            name: "Petrovich",
            towns: ""
        },
        mainRows: ["name", "rating"]
    }
    const component = create(<ListItem {...mockProps}/>);
    expect(component.toJSON()).toMatchSnapshot();
});