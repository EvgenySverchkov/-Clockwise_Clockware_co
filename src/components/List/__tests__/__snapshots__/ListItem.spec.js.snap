// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <ListHeader /> 1`] = `
Array [
  <td
    style={
      Object {
        "wordWrap": "break-word",
      }
    }
  >
    Petrovich
  </td>,
  <td
    style={
      Object {
        "wordWrap": "break-word",
      }
    }
  >
    5
  </td>,
]
`;
